plugins/l4lb/l4lb.go

```diff
@@ -0,0 +1,281 @@
+	ipamResult, err := ipam.ExecAdd(spartan.Config.IPAM.Type, spartanNetConf)
+	if err != nil {
+		return err
+	}
+
+	result, err := current.NewResultFromResult(ipamResult)
+	if err != nil {
+		return err
+	}
+
+	if result.IPs == nil {
+		return errors.New("IPAM plugin returned missing IPv4 config")
+	}
+
+	// Make sure we got only one IP and that it is IPv4
+	if (len(result.IPs) > 1) || result.IPs[0].Address.IP.To4() == nil {
+		return errors.New("Expecting a single IPv4 address from IPAM")
+	}
+
+	hostVethName, err := setupContainerVeth(args.Netns, spartan.Config.Interface, conf.MTU, result, spartan.IPs)
+	if err != nil {
+		return err
+	}
+
+	hostVeth, err := netlink.LinkByName(hostVethName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup host VETH %q: %v", hostVethName, err)
+	}
+
+	containerRoute := netlink.Route{
+		LinkIndex: hostVeth.Attrs().Index,
+		Dst: &net.IPNet{
+			IP:   result.IPs[0].Address.IP,
+			Mask: net.IPv4Mask(0xff, 0xff, 0xff, 0xff),
+		},
+		Scope: netlink.SCOPE_LINK,
+	}
+
+	if err = netlink.RouteAdd(&containerRoute); err != nil {
+		return fmt.Errorf("failed to add spartan route %v: %v", containerRoute, err)
+	}
+
+	//TODO(asridharan): We probably need to update the DNS result to
+	//make sure that we override the DNS resolution with the spartan
+	//network, since the operator has explicitly requested to use the
+	//spartan network.
+
+	// We always return the result from the delegate plugin and not from
+	// this plugin.
+	return delegateResult.Print()
+}
+
+func cmdDel(args *skel.CmdArgs) error {
+	conf := NetConf{}
+	if err := json.Unmarshal(args.StdinData, &conf); err != nil {
+		return fmt.Errorf("failed to load netconf: %v", err)
+	}
+
+	spartanNetConf, err := json.Marshal(spartan.Config)
+	if err != nil {
+		return fmt.Errorf("failed to marshall the `spartan-network` IPAM configuration: %v", err)
+	}
+
+	if err = ipam.ExecDel(spartan.Config.IPAM.Type, spartanNetConf); err != nil {
+		return err
+	}
+
+	if args.Netns == "" {
+		return nil
+	}
+
+	// Ideally, the kernel would clean up the veth and routes within the
+	// network namespace when the namespace is destroyed. We are still
+	// explicitly deleting the interface here since we want the IP
+	// address associated with the interface. We will then use the IP
+	// address to clean up any associated routes in the host network
+	// namespace. We also don't want any interfaces to be
+	// present when the delegate plugin is invoked, since the presence
+	// of these routes might confuse the delegate plugin.
+	var ipn *net.IPNet
+	err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+		// We just need to delete the interface, the associated routes
+		// will get deleted by themselves.
+		var err error
+		ipn, err = ip.DelLinkByNameAddr(spartan.Config.Interface, netlink.FAMILY_V4)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to delete spartan interface in container: %v", err)
```

Comment:
if the interface will get deleted by itself we don't need to delete it here then?